enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,995 @@
+	FailureDomain *string `json:"failureDomain,omitempty"`
+}
+```
+
+### CCCMO
+
+The CCCMO will need to be modified to support the VSpherePlatformSpec.
+It will need to be able to use the existing INI-based cloud-config spec
+and also translate that configuration to YAML-based cloud-config.
+
+### installer
+
+#### Platform Spec
+
+The platform spec needs to be modified to support our initial goals of multiple
+datacenters (regions) and clusters (zones) and vcenters. This platform spec
+design is based on the changes suggested in openshift/api, Cluster API for vSphere
+and out-of-tree CCM.
+
+We are adding multiple additional parameters to the Platform struct:
+
+- `VCenters`
+- `DeploymentZones`
+- `FailureDomains`
+
+`VCenters` will contain the connections and configuration for each vCenter
+that is required for the out-of-tree CCM. Note: Only a single vCenter will
+be supported by this effort. While the out-of-tree CCM supports multiple vCenters
+the out-of-tree CSI does not.
+
+`FailureDomains` will define the configuration of a region, zone, topology and
+the topology type. `Topology` defines the vCenter objects that make up a region or zone
+including `Datacenter`, `ComputeCluster`, `Hosts`, `Networks` and a `Datastore`.
+
+`DeploymentZones` will define the association between a FailureDomain, vcenter and
+PlacementConstraints - which include Resource Pools and Folders.
+
+The existing platform spec vcenter parameters will be deprecated
+but _not_ removed or remove support for using those parameters.
+
+```golang
+// VSpherePlatformFailureDomainType is the name of the failure domain type.
+// There are two defined failure domains currently, Datacenter and ComputeCluster.
+// Each represents a vCenter object type within a vSphere environment.
+// +kubebuilder:validation:Enum="";HostGroup;Datacenter;ComputeCluster
+type FailureDomainType string
+
+const (
+    // HostGroup as a type allows the use of a group of ESXi hosts
+    // to be represented as a failure domain zone. When using this
+    // case it is expected that region would be a cluster.
+    // HostGroups within vCenter must be preconfigured and
+    // assigned in the topology.
+    HostGroupFailureDomain      VSpherePlatformFailureDomainType = "HostGroup"
+
+    // ComputeCluster failure domain can either be a zone or region.
+    // The vCenter cluster is required to preconfigured and
+    // assigned in the topology.
+    ComputeClusterFailureDomain VSpherePlatformFailureDomainType = "ComputeCluster"
+
+    // Datacenter failure domain can be only a region. The vcenter
+    // datacenter is required to be preconfigred and assigned
+    // in the topology. If used the zone would be of type ComputeCluster.
+    DatacenterFailureDomain     VSpherePlatformFailureDomainType = "Datacenter"
+)
+
+// FailureDomainSpec holds the region and zone failure domain and
+// the vCenter topology of that failure domain.
+type FailureDomainSpec struct {
```

Comment:
In the installer we don't typically use the `Spec` nomenclature. What is the difference between the `FailureDomainSpec` and the `FailureDomain` types defined here? Can we just combine them?